deploy/role.yaml

```diff
@@ -41,6 +41,20 @@ rules:
 - apiGroups:
   - metal3.io
   resources:
-  - '*'
+  - baremetalhosts
```

Comment:
If you are trying to apply the least privilege approach, you should also restrict the access to other resources defined above, to the minimum needed. for example restrict the verb to get and list for secrets and configmaps, and restrict the resources accessed to secrets and configmaps + other needed things like events etc.